sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c

```diff
@@ -213,6 +339,218 @@ void gnrc_ipv6_nib_rtr_adv_pio_cb(gnrc_netif_t *upstream, const ndp_opt_pi_t *pi
+ */
+static bool _all_zero(const uint8_t *addr, size_t len)
+{
+    for (const uint8_t *end = addr + len; addr != end; ++addr) {
+        if (*addr) {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * @brief Allocates a l2 address in the `l2addrs` array
+ *
+ * @param[in]   addr    The l2 address to insert
+ * @param[in]   len     l2 address length
+ *
+ * @return  1 if the address was added to the `l2addrs` array
+ *          0 if the address was already in the array
+ *         -1 if there was no more space in the `l2addrs` array
+ */
+static int _alloc_l2addr_entry(const void *addr, size_t len)
+{
+    int empty = -1;
+    for (unsigned i = 0; i < CONFIG_GNRC_IPV6_AUTO_SUBNETS_PEERS_MAX; ++i) {
+        if (_all_zero(l2addrs[i], len)) {
+            empty = i;
+            continue;
+        }
+        if (memcmp(addr, l2addrs[i], len) == 0) {
+            return 0;
+        }
+    }
+
+    if (empty < 0) {
+        return -1;
+    }
+
+    memcpy(l2addrs[empty], addr, len);
+    return 1;
+}
+
+/**
+ * @brief Compare the l2 address of the received packet with the l2 address of the
+ *        interface it was received on.
+ *
+ *        Only the first packet from a host generates a comparison, all subsequent
+ *        packets will be ignored until the `l2addrs` array is reset.
+ *
+ * @param[in] upstream interface, ignore if the source does not match
+ * @param[in] pkt   a received packet
+ *
+ * @return  1 if the sender l2 address is in order before the local l2 address
+ * @return  0 if the order could not be determined or a packet from the sender
+ *            was already processed
+ * @return -1 if the sender l2 address is in order behind the local l2 address
+ */
+static int _get_my_l2addr_rank(gnrc_netif_t *iface, gnrc_pktsnip_t *pkt)
+{
+    const void *src_addr;
+    gnrc_pktsnip_t *netif_hdr;
+    gnrc_netif_hdr_t *hdr;
+
+    if (iface == NULL) {
+        return 0;
+    }
+
+    netif_hdr = gnrc_pktsnip_search_type(pkt, GNRC_NETTYPE_NETIF);
+    if (netif_hdr == NULL) {
+        return 0;
+    }
+
+    /* ignore packet if it was received on the wrong interface */
+    hdr = netif_hdr->data;
+    if (iface->pid != hdr->if_pid) {
+        return 0;
+    }
+
+    /* ignore packets without source address */
+    src_addr = gnrc_netif_hdr_get_src_addr(hdr);
+    if (src_addr == NULL) {
+        return 0;
+    }
+
+    /* check if we have seen the host before */
+    if (_alloc_l2addr_entry(src_addr, iface->l2addr_len) == 0) {
+        return 0;
+    }
+
+    return memcmp(iface->l2addr, src_addr, iface->l2addr_len);
+}
+
+static void *_eventloop(void *arg)
+{
+    (void)arg;
+
+    xtimer_t timeout_timer;
+    msg_t msg, timeout_msg = { .type = SERVER_MSG_TYPE_TIMEOUT };
+    gnrc_netreg_entry_t server = GNRC_NETREG_ENTRY_INIT_PID(0, KERNEL_PID_UNDEF);
+    uint8_t idx_start = 0;
+    uint8_t subnets = gnrc_netif_numof() - 1;
+    uint8_t resend_tx = CONFIG_GNRC_IPV6_AUTO_SUBNETS_TX_RETRIES;
```

Comment:
```suggestion
    uint8_t tx_retries = CONFIG_GNRC_IPV6_AUTO_SUBNETS_TX_RETRIES;
```

(or rather, align with whatever name `CONFIG_GNRC_IPV6_AUTO_SUBNETS_TX_RETRIES` ends up with)